diff -uNr luagit2-master/CMakeLists.txt luagit2-master_patched/CMakeLists.txt
--- luagit2-master/CMakeLists.txt	2019-06-25 16:05:14.000000000 +0200
+++ luagit2-master_patched/CMakeLists.txt	2020-09-16 23:52:15.179687138 +0200
@@ -5,8 +5,8 @@
 
 MESSAGE("${CMAKE_MODULE_PATH}")
  
-find_package(Lua 5.3 EXACT REQUIRED)
-find_package(LibGit2 REQUIRED)
+#find_package(Lua 5.3 EXACT REQUIRED)
+#find_package(LibGit2 REQUIRED)
  
 include_directories (
     ${CMAKE_CURRENT_BINARY_DIR}
@@ -65,11 +65,11 @@
              ${src_dest}/main_wrapper.c 
  )
 
-set(lib_dest "lib/lua/5.3")
+set(lib_dest "lua_plugins")
 
 OPTION(BUILD_SHARED_LIBS ON)
  
-add_library (${PROJECT_NAME} SHARED ${SOURCES} ${LUA_LIBRARIES})
-target_link_libraries (${PROJECT_NAME} lua5.3 git2)
+add_library (${PROJECT_NAME} SHARED ${SOURCES})
+target_link_libraries (${PROJECT_NAME} ${LUA_LIBRARIES} ${LIBGIT2_LIBRARY} ${LIBSSL_LIBRARY} ${LIBCRYPTO_LIBRARY} ${LIBSSH2_LIBRARY} ${LIBGCRYPT_LIBRARY} ${LIBGPG_ERROR_LIBRARY} pthread)
 set_target_properties (${PROJECT_NAME} PROPERTIES PREFIX "")
 install(TARGETS ${PROJECT_NAME} DESTINATION "${lib_dest}" )
