diff -uNr libgit2-1.0.1/src/describe.c libgit2-1.0.1_patched/src/describe.c
--- libgit2-1.0.1/src/describe.c	2020-06-04 10:46:58.000000000 +0200
+++ libgit2-1.0.1_patched/src/describe.c	2020-09-16 20:59:02.219302950 +0200
@@ -685,7 +685,7 @@
 			get_name, &data)) < 0)
 				goto cleanup;
 
-	if (git_oidmap_size(data.names) == 0 && !opts->show_commit_oid_as_fallback) {
+	if (git_oidmap_size(data.names) == 0 && !normalized.show_commit_oid_as_fallback) {
 		git_error_set(GIT_ERROR_DESCRIBE, "cannot describe - "
 			"no reference found, cannot describe anything.");
 		error = -1;
