diff -uNr libgit2-1.0.1/CMakeLists.txt libgit2-1.0.1_patched/CMakeLists.txt
--- libgit2-1.0.1/CMakeLists.txt	2020-06-04 10:46:58.000000000 +0200
+++ libgit2-1.0.1_patched/CMakeLists.txt	2020-09-16 23:33:20.359945034 +0200
@@ -219,9 +219,7 @@
 
 	IF (MINGW OR MSYS) # MinGW and MSYS always do PIC and complain if we tell them to
 		STRING(REGEX REPLACE "-fPIC" "" CMAKE_SHARED_LIBRARY_C_FLAGS "${CMAKE_SHARED_LIBRARY_C_FLAGS}")
-	ELSEIF (BUILD_SHARED_LIBS)
-		ADD_C_FLAG_IF_SUPPORTED(-fvisibility=hidden)
-
+	ELSE ()
 		SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 	ENDIF ()
 
